api_fetch.js

```diff
@@ -131,12 +148,36 @@ class ChallengeListView {
   }
 }
 
-// Show All Challenge cards on Challenges page
-const challengesDiv = document.querySelector("#challenges__container");
+// Filter out the top3 challenges by rating
+class Top3View {
+  async render(container) {
+    const api = new APIadapter();
+    const challenges = await api.getAllChallenges();
+    const sorted = challenges.sort(
+      (item0, item1) => item1.data.rating - item0.data.rating
+    );
+    for (let i = 0; i < 3; i++) {
+      const challenge = sorted[i];
+      const element = challenge.render();
+      container.append(element);
+    }
+  }
+}
+
+// Look for which page is showing and render all challenges or top3
+const currentPage = document.body.dataset.page;
```

Comment:
Nice use of `data-` attributes and `dataset`! For things like this, the HTML `class` attribute is often used, but this is valid as well and perhaps even better if the value is only relevant to JS (not CSS).